packages/filesystem/src/browser/filesystem-watcher.spec.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as temp from 'temp';
+import * as chai from 'chai';
+import * as fs from 'fs-extra';
+import * as assert from 'assert';
+import * as chaiAsPromised from 'chai-as-promised';
+import URI from "@theia/core/lib/common/uri";
+import { FileUri } from "@theia/core/lib/node";
+import { PreferenceService, PreferenceServer, PreferenceClient } from "@theia/preferences-api";
+import { FileSystemWatcher } from './filesystem-watcher';
+import { createFileSystemPreferences } from './filesystem-preferences';
+import { NsfwFileSystemWatcherServer } from '../node/nsfw-watcher/nsfw-filesystem-watcher';
```

Comment:
let's make it NsfwFileSystemWatcherServer test

Reply:
I've just added  `nsfw-filesystem-watcher.spec.ts` tests, let me know if that's what you had in mind!